include/seqan3/std/bit

```diff
@@ -49,3 +74,146 @@ enum class endian
 } //namespace std
 
 #endif // __cpp_lib_endian
+
+#if !defined(SEQAN3_CPP_LIB_INT_POW2) || !defined(SEQAN3_CPP_LIB_BITOPS)
+
+namespace std::detail
+{
+/*!\brief How many bits has a type?
+ * \ingroup bit
+ * \tparam type_t The type to determine the number of bits.
+ */
+template <typename type_t>
+constexpr auto sizeof_bits = CHAR_BIT * sizeof(type_t);
+} // namespace std::detail
+
+#endif // !defined(SEQAN3_CPP_LIB_INT_POW2) || !defined(SEQAN3_CPP_LIB_BITOPS)
+
+#ifndef SEQAN3_CPP_LIB_INT_POW2
+
+namespace std
+{
+
+#ifndef SEQAN3_CPP_LIB_BITOPS
+// forward declare
+template<class T> constexpr int countl_zero(T x) noexcept;
+#endif // SEQAN3_CPP_LIB_BITOPS
+
+// // bit_­cast
+// template<class To, class From> constexpr To bit_cast(const From& from) noexcept;
+//
+// // integral powers of 2
```

Comment:
what's this?

Reply:
It shows what different things are defined within the bit header and makes it easier, if someone wants to use that feature and implements it in the shim header.